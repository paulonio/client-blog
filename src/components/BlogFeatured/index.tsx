import React from 'react';
import Link from 'next/link';
import { useTranslation } from 'next-i18next';

import Button from '../Button';

import { POSTS } from '@/constants/posts';

import { sen } from '@/styles/fonts';
import styles from './styled.module.scss';
import { stringToKey } from '@/utils/utils';

const post = POSTS[1];

const BlogFeatured = () => {
  const { id, username, date, title, body, urlToImage } = post;
  const { t } = useTranslation(['blog', 'common']);

  return (
    <section className={styles.blog}>
      <div className="container">
        <div className={styles.blog__wrapper}>
          <div className={styles.blog__featured}>
            <h5>{t('heading.subtitle')}</h5>
            <h2 className={sen.className} data-testid="blog-featured-title">
              {t(stringToKey(title))}
            </h2>
            <p className="body-l">
              <span className={styles.name}>{t('heading.author')}</span> |{' '}
              {t(stringToKey(date), { ns: 'common' })}
            </p>
            <p className="body-s">{t(stringToKey(body))}</p>
            <Link href={`/blog/${id}`}>
              <Button>
                {t('heading.read')} {'>'}
              </Button>
            </Link>
          </div>
          <div>
            <img src={urlToImage} alt={username} />
          </div>
        </div>
      </div>
    </section>
  );
};

export default BlogFeatured;
