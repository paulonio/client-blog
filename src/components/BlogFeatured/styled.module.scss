@import '../../styles/sass/constants';

.blog {
  background-color: $color-light-m;

  &__wrapper {
    padding: 80px 0;
    display: flex;
    align-items: center;
    justify-content: space-between;
    gap: 141px;

    @media screen and (max-width: 1000px) {
      padding: 10px 0;
      flex-direction: column-reverse;
      gap: 24px;

      & div {
        width: 80%;
      }

      & img {
        width: 100%;
        height: 100%;
      }
    }
  }

  &__featured {
    display: flex;
    flex-direction: column;
    align-items: start;
    row-gap: 16px;

    & h5 {
      margin-bottom: 4px;
    }

    & a {
      margin-top: 16px;
    }
  }
}

.name {
  color: $color-secondary;
}