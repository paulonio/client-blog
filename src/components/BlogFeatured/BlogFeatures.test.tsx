import React from 'react';
import { render, screen } from '@testing-library/react';
import BlogFeatured from '.';

describe('Blog featured', () => {
  it('should render blog featured component', () => {
    const { container } = render(<BlogFeatured />);

    expect(container).toMatchSnapshot();
  });

  it('should render blog featured component', () => {
    render(<BlogFeatured />);

    const title = screen.getByTestId('blog-featured-title');

    expect(title).toBeInTheDocument();
  });
});
