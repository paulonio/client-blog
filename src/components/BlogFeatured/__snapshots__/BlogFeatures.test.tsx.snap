// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog featured should render blog featured component 1`] = `
<div>
  <section
    class="blog"
  >
    <div
      class="container"
    >
      <div
        class="blog__wrapper"
      >
        <div
          class="blog__featured"
        >
          <h5>
            heading.subtitle
          </h5>
          <h2
            class="className"
            data-testid="blog-featured-title"
          >
            step-by-step-guide
          </h2>
          <p
            class="body-l"
          >
            <span
              class="name"
            >
              heading.author
            </span>
             |
             
            aug-23,
          </p>
          <p
            class="body-s"
          >
            duis-aute
          </p>
          <a
            href="/blog/2"
          >
            <button
              class="button"
              type="button"
            >
              heading.read
               
              &gt;
            </button>
          </a>
        </div>
        <div>
          <img
            alt="Robert Fox"
            src="/assets/images/blog-featured.jpg"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
